### PR TITLE
22494 update furnishing document style

### PR DESCRIPTION
**Pull request by kzdev420**
*Issue #:* /bcgov/entity#22494

*Description of changes:*

- Color version
Screenshot 2024-08-13 at 2 16 41 PM

- Greyscale version
Screenshot 2024-08-13 at 2 21 22 PM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the lear license (Apache 2.0).

### CODE REVIEW

**Comment by chenhongjing** [comment]
The following is another reqquirement I got from BCMail+ when I worked on the document.  Just want to make sure your changes will not affect any address & folding position so that it fits into the window envelope.

image

**Comment by argush3** [comment]
So it doesn't look like you have made updates to the letter specific to the EP flow.

One thing missing I can see from a quick look at the designs are the following:

image

The business identifier you tested with...was it an EP?

**Comment by kzdev420** [comment]
`api/v2/businesses/A0883551/furnishings/218/document`

[response-ep.pdf](https://github.com/user-attachments/files/16622350/response-ep.pdf)

**Comment by kzdev420** [comment]
> So it doesn't look like you have made updates to the letter specific to the EP flow.
> 
> One thing missing I can see from a quick look at the designs is the following:
> 
> image
> 
> The business identifier you tested with...was it an EP?
> 
> Also, Hongjing had added tests for the BC company furnishing documents in her [PR](https://github.com/bcgov/lear/pull/2849/files#diff-665af9c4e4e6a118484d095351188d09d76b9b2243dce8ffa29d39a48565dde4). You should see if you can add or update tests to test the EP flow.

Updated

**Comment by argush3** [comment]
Can you upload the black and white version of the letter too?

**Comment by kzdev420** [comment]
> Can you upload the black and white version of the letter too?

Screenshot 2024-08-15 at 9 35 53 AM

**Comment by chenhongjing** [comment]
The wording in cancellation letter is different and you have to check and do some changes too. Besides, it will be good if you can show the measurement for greyscale version of letter because the address & folding position is more important for this version.

**Comment by kzdev420** [comment]
> The wording in cancellation letter is different and you have to check and do some changes too. Besides, it will be good if you can show the measurement for greyscale version of letter because the address & folding position is more important for this version.

Updated
[response-ep.pdf](https://github.com/user-attachments/files/16640691/response-ep.pdf)
[response.pdf](https://github.com/user-attachments/files/16640693/response.pdf)
Screenshot 2024-08-16 at 12 01 08 PM

**Comment by argush3** [comment]
> > The wording in cancellation letter is different and you have to check and do some changes too. Besides, it will be good if you can show the measurement for greyscale version of letter because the address & folding position is more important for this version.
> 
> Updated [response-ep.pdf](https://github.com/user-attachments/files/16640691/response-ep.pdf) [response.pdf](https://github.com/user-attachments/files/16640693/response.pdf) Screenshot 2024-08-16 at 12 01 08 PM

can you attach the final color versions for BC and EP as well?

**Comment by kzdev420** [comment]
> > > The wording in cancellation letter is different and you have to check and do some changes too. Besides, it will be good if you can show the measurement for greyscale version of letter because the address & folding position is more important for this version.
> > 
> > 
> > Updated [response-ep.pdf](https://github.com/user-attachments/files/16640691/response-ep.pdf) [response.pdf](https://github.com/user-attachments/files/16640693/response.pdf) Screenshot 2024-08-16 at 12 01 08 PM
> 
> can you attach the final color versions for BC and EP as well?


[response-c-ep.pdf](https://github.com/user-attachments/files/16641094/response-c-ep.pdf)

[response-c.pdf](https://github.com/user-attachments/files/16641095/response-c.pdf)


Screenshot 2024-08-16 at 12 55 45 PM

**Comment by kzdev420** [comment]
Screenshot 2024-08-14 182725

Screenshot 2024-08-16 160522

File: legal-api/tests/unit/reports/test_report_v2.py [hunk 1]
```diff
@@ -21,14 +21,15 @@
...
 
 @pytest.mark.parametrize(
-    'test_name, variant', [
-        ('COMMENCEMENT_DEFAULT', 'default'),
-        ('COMMENCEMENT_GREYSCALE', 'greyscale'),
+    'test_name, variant, legal_type', [
+        ('COMMENCEMENT_DEFAULT_DISSOLUTION', 'default', 'BEN'),
+        ('COMMENCEMENT_GREYSCALE_DISSOLUTION', 'greyscale', 'BC'),
+        ('COMMENCEMENT_GREYSCALE_CANCELLATION', 'greyscale', 'A'),
     ]
```
Comment (argush3) [inline legal-api/tests/unit/reports/test_report_v2.py:28]: should add a test for default(color) EP scenario

File: legal-api/src/legal_api/reports/report_v2.py [hunk 2]
```diff
@@ -148,12 +148,23 @@ def _format_furnishing_data(self):
             processed_date = LegislationDatetime.as_legislation_timezone(self._report_date_time)
         self._report_data['furnishing']['processedDate'] = processed_date.strftime(OUTPUT_DATE_FORMAT)
 
+        self._report_data['furnishing']['businessLegalType'] = 'BC'
+        if self._business.legal_type in [Business.LegalTypes.EXTRA_PRO_A.value]:
+            self._report_data['furnishing']['businessLegalType'] = 'EP'
```
Comment (argush3) [inline legal-api/src/legal_api/reports/report_v2.py:153]: I think we should keep using `A` as the legal type value for EPs.
Do we actually have a need distinguish BC companies in the letter templates?

If not, I think we can replace all 3 lines with
i.e `self._report_data['furnishing']['businessLegalType'] =  self._business.legal_type`

From a quick look at `legal-api/report-templates/noticeOfDissolutionCommencement.html`, I think all you need is ` if furnishing.businessLegalType == 'A'`

Comment (kzdev420) [reply legal-api/src/legal_api/reports/report_v2.py]: Updated

File: legal-api/tests/unit/reports/test_report_v2.py (hunk 1 above)
Comment (kzdev420) [reply legal-api/tests/unit/reports/test_report_v2.py]: Added

**Review by argush3 — APPROVED** [review]
approved assuming tests pass CI

**Review by chenhongjing — APPROVED** [review]

**Pull request merged**